apps/preview/next/pages/showcases/Gallery/GalleryVertical.tsx

```diff
@@ -0,0 +1,148 @@
+  { imageSrc: Images.gallery3Src, alt: 'backpack3' },
+  { imageSrc: Images.gallery4Src, alt: 'backpack4' },
+  { imageSrc: Images.gallery5Src, alt: 'backpack5' },
+  { imageSrc: Images.gallery6Src, alt: 'backpack6' },
+  { imageSrc: Images.gallery7Src, alt: 'backpack7' },
+  { imageSrc: Images.gallery8Src, alt: 'backpack8' },
+  { imageSrc: Images.gallery9Src, alt: 'backpack9' },
+  { imageSrc: Images.gallery10Src, alt: 'backpack10' },
+];
+
+const thumbImages = [
+  { imageSrc: Images.gallery1ThumbSrc, alt: 'backpack1' },
+  { imageSrc: Images.gallery2ThumbSrc, alt: 'backpack2' },
+  { imageSrc: Images.gallery3ThumbSrc, alt: 'backpack3' },
+  { imageSrc: Images.gallery4ThumbSrc, alt: 'backpack4' },
+  { imageSrc: Images.gallery5ThumbSrc, alt: 'backpack5' },
+  { imageSrc: Images.gallery6ThumbSrc, alt: 'backpack6' },
+  { imageSrc: Images.gallery7ThumbSrc, alt: 'backpack7' },
+  { imageSrc: Images.gallery8ThumbSrc, alt: 'backpack8' },
+  { imageSrc: Images.gallery9ThumbSrc, alt: 'backpack9' },
+  { imageSrc: Images.gallery10ThumbSrc, alt: 'backpack10' },
+];
+
+export default function GalleryVertical() {
+  const lastThumbRef = useRef<HTMLButtonElement>(null);
+  const thumbsRef = useRef<HTMLDivElement>(null);
+  const firstThumbRef = useRef<HTMLButtonElement>(null);
+  const [activeIndex, setActiveIndex] = useState(0);
+
+  const firstThumbVisible = useIntersection(firstThumbRef, {
+    root: thumbsRef.current,
+    rootMargin: '0px',
+    threshold: 1,
+  });
+
+  const lastThumbVisible = useIntersection(lastThumbRef, {
+    root: thumbsRef.current,
+    rootMargin: '0px',
+    threshold: 1,
+  });
+
+  const onDragged = (event: SfScrollableOnDragEndData) => {
+    if (event.swipeRight && activeIndex > 0) {
+      setActiveIndex((currentActiveIndex) => currentActiveIndex - 1);
+    } else if (event.swipeLeft && activeIndex < images.length - 1) {
+      setActiveIndex((currentActiveIndex) => currentActiveIndex + 1);
+    }
+  };
+
+  return (
+    <div className="relative max-h-[600px] flex h-full">
+      <SfScrollable
+        ref={thumbsRef}
+        className="items-center w-full [&::-webkit-scrollbar]:hidden [-ms-overflow-style:none] [scrollbar-width:none]"
+        direction="vertical"
+        activeIndex={activeIndex}
+        previousDisabled={activeIndex === 0}
+        nextDisabled={activeIndex === images.length - 1}
+        slotPreviousButton={
+          <SfButton
+            className={classNames(
+              'absolute !rounded-full z-10 top-4 rotate-90 bg-white',
+              firstThumbVisible?.isIntersecting ? 'hidden' : null,
+            )}
+            variant="secondary"
+            size="sm"
+            square
+            slotPrefix={<SfIconChevronLeft />}
+          />
+        }
+        slotNextButton={
+          <SfButton
+            className={classNames(
+              'absolute !rounded-full z-10 bottom-4 rotate-90 bg-white',
+              lastThumbVisible?.isIntersecting ? 'hidden' : null,
```

Comment:
https://github.com/JedWatson/classnames#usage
```suggestion
              hidden: lastThumbVisible?.isIntersecting,
```